WalletWasabi/Services/WalletService.cs

```diff
@@ -468,13 +472,13 @@ private void ProcessTransaction(SmartTransaction tx, List<HdPubKey> keys = null)
 				if (foundKey != default)
 				{
 					foundKey.SetKeyState(KeyState.Used, KeyManager);
-					List<SmartCoin> spentOwnCoins = Coins.Where(x => tx.Transaction.Inputs.Any(y => y.PrevOut.Hash == x.TransactionId && y.PrevOut.N == x.Index)).ToList();
+					List<SmartCoin> spentOwnCoins = Coins.Where(x => tx.Transaction.Inputs.Any(y => y.PrevOut.N == x.Index && y.PrevOut.Hash == x.TransactionId)).ToList();
```

Comment:
ACK order change

Reply:
This 4-5 lines are the last to review, before I merge, and I am looking at it for over half an hour and I'm still not sure. Such complicated code. Anyway I must go now, will get back to it later.